File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2022s/ID06IO2/Strategy/Command/SportsmenCommandTrening.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Strategy
{
    class SportsmenCommandTrening: SportsmenCommandComplex
    {
        private readonly List<SportsmenCommand> cmds;

        public SportsmenCommandTrening(string name,List<SportsmenCommand> cmds):base(name)
        {
            this.cmds= cmds;
        }

        public override string execute(Sportsmen sportsmen)
        {
            StringBuilder sb = new StringBuilder();
            bool first = true;
            foreach (SportsmenCommand cmd in cmds)
            {
                if (first) first = false; else sb.Append(", ");
                sb.Append(cmd.execute(sportsmen));                
            }
            return sb.ToString();
        }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022s/IZ08IO3/Commands/src/SportsmenCommandTrening.java

import java.util.List;

public class SportsmenCommandTrening extends SportsmenCommandComplex
{
   private final List<SportsmenCommand> cmds;
   public SportsmenCommandTrening(String name,List<SportsmenCommand> cmds)
   {
      super(name);
      this.cmds=cmds;
   }
   @Override public String execute(Sportsmen s)
   {
      StringBuilder sb=new StringBuilder().append('{');
      for(int i=0;i<cmds.size();++i)
      {
         if(i>0) sb.append(", ");
         sb.append(cmds.get(i).execute(s));
      }
      return sb.append('}').toString();
   }
}
